### PR TITLE
Handle long descriptions in detailsview

### PR DESCRIPTION
**Pull request by nullxDEADBEEF**
### Is your feature request related to a problem? Please describe.

Right now the detailsview descriptions can get very long and out of hand. 

image

https://streamchaser.tv/person/18918

### Describe the solution you'd like

If like the description length was fixed, and you could simply press `show more` to get the full description

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### CODE REVIEW

**Comment by nullxDEADBEEF** [comment]
Screenshot 2021-11-08 at 00 41 33

File: frontend/src/routes/person/[id].svelte [hunk 1]
```diff
@@ -11,9 +11,11 @@
     const PERSON_DETAIL_URL: string = `${variables.apiPath}/person/${$page.params.id}`;
     const IMG_URL: string = 'https://image.tmdb.org/t/p/original/';
     const LOW_RES_IMG_URL: string = 'https://image.tmdb.org/t/p/w500/';
+    const INITIAL_BIOGRAPHY_LENGTH: number = 200;
```
Comment (AndreasPB) [inline frontend/src/routes/person/[id].svelte:14]: I would probably put in a larger number until we have more things to show, it feels a little empty(around ~500-ish)

**Review by AndreasPB — APPROVED** [review]
Approved with the one change :shipit:

**Comment by Primdahl26** [comment]
Hmm do we maybe want something like this? 

image

@streamchaser/developers

**Comment by AndreasPB** [comment]
Yeah, that's a good idea. We are probably overusing that one button/feature a bit :D

**Review by AndreasPB — CHANGES_REQUESTED** [review]
What @Primdahl26 said :D

File: frontend/src/routes/person/[id].svelte [hunk 2]
```diff
@@ -64,7 +66,19 @@
                     </figure>
                     <div class="max-w-md card-body">
                         <h2 class="card-title">{person.name}</h2>
-                        <p>{person.biography}</p>
+                        {#if currentBiographyLength <= INITIAL_BIOGRAPHY_LENGTH && person.biography.length > INITIAL_BIOGRAPHY_LENGTH}
+                            <p>{person.biography.slice(0, currentBiographyLength)}...</p>
+                            <div class="flex space-x-1 justify-center p-1">
+                                <button
+                                    on:click={() => currentBiographyLength = person.biography.length}
+                                    type="button"
```
Comment (Primdahl26) [inline frontend/src/routes/person/[id].svelte:74]: Maybe something else than a button? A link like shown a bit higher in the issue

**Review by Primdahl26 — CHANGES_REQUESTED** [review]

**Comment by nullxDEADBEEF** [comment]
Screenshot 2021-11-18 at 10 17 26

File: frontend/src/routes/person/[id].svelte (hunk 2 above)
Comment (nullxDEADBEEF) [reply frontend/src/routes/person/[id].svelte]: See latest image, for the newest change. I will add this on movie and tv as well :D

File: frontend/src/routes/person/[id].svelte (hunk 1 above)
Comment (nullxDEADBEEF) [reply frontend/src/routes/person/[id].svelte]: Put on 500 :D

File: frontend/src/components/read_more.svelte [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<script lang="ts">
+    export let currentDescriptionLength;
+    export let mediaDescription;
+    export let initialDescriptionLength;
+</script>
+
+{#if currentDescriptionLength <= initialDescriptionLength && mediaDescription.length > initialDescriptionLength}
+    <p>{mediaDescription.slice(0, currentDescriptionLength)}
+        <a href="" class="cursor-pointer"
```
Comment (Primdahl26) [inline frontend/src/components/read_more.svelte:9]: Can we make the `... read more` a other color like blue?

`<a href="" class="cursor-pointer text-blue-400"`

**Review by Primdahl26 — CHANGES_REQUESTED** [review]
Very nice, only the color thing i would like changed

**Comment by Primdahl26** [comment]
image

https://streamchaser.tv/tv/61222 has a very interesting long text on the cast actors. Should we handle this in this PR or another?

File: frontend/src/components/read_more.svelte (hunk 3 above)
Comment (nullxDEADBEEF) [reply frontend/src/components/read_more.svelte]: Implemented in the newest commit

**Review by Primdahl26 — APPROVED** [review]

**Pull request merged**